Firestore/core/test/unit/credentials/firebase_app_check_credentials_provider_test.mm

```diff
@@ -178,6 +178,44 @@ - (nonnull NSString*)tokenDidChangeNotificationName {
   ASSERT_EQ(std::future_status::ready, token_future.wait_for(kTimeout));
 }
 
+// Regression test for https://github.com/firebase/firebase-ios-sdk/issues/8895
+TEST(FirebaseAppCheckCredentialsProviderTest,
+     ListenForTokenChangesIgnoresUnrelatedNotifcations) {
+  auto token_promise = std::make_shared<std::promise<std::string>>();
+
+  FIRApp* app = testutil::AppForUnitTesting();
+  FSTAppCheckFake* app_check = [[FSTAppCheckFake alloc] initWithToken:@""];
+  FirebaseAppCheckCredentialsProvider credentials_provider(app, app_check);
```

Comment:
Can we add a variant of this test and `GetToken` test with `app_check == nil`?

Reply:
There is already a test for this: GetTokenNoProvider.